src/mathf/dimensions-2d.ts

```diff
@@ -0,0 +1,60 @@
+import { Vector } from './vector';
+import { SCROLL_ELEMENT } from '../dom/scroll-element';
+import { ROOT_ELEMENT } from '../dom/root-element';
+
+export class Dimensions2d extends Vector {
+  get width(): number {
+    return this.x;
+  }
+
+  get height(): number {
+    return this.y;
+  }
+
+  static fromCanvas<T extends Dimensions2d>(
+      element: HTMLCanvasElement = null
+  ): T {
+    return <T>new this(element.width, element.height);
+  }
+
+  static fromVideo<T extends Dimensions2d>(
+      element: HTMLVideoElement = null
+  ): T {
+    return <T>new this(element.videoWidth, element.videoHeight);
+  }
+
+  static fromElementOffset<T extends Dimensions2d>(
+      element: HTMLElement = null
+  ): T {
+    if (element) {
+      return <T>new this(element.offsetWidth, element.offsetHeight);
+    } else {
+      return this.fromInnerWindow();
+    }
+  }
+
+  static fromRootElement<T extends Dimensions2d>() {
+    return <T>new this(ROOT_ELEMENT.clientWidth, ROOT_ELEMENT.clientHeight);
+  }
+
+  static fromScrollElementClient<T extends Dimensions2d>() {
+    return <T>new this(SCROLL_ELEMENT.clientWidth, SCROLL_ELEMENT.clientHeight);
+  }
+
+  static fromInnerWindow<T extends Dimensions2d>() {
+    return <T>new this(window.innerWidth, window.innerHeight);
```

Comment:
I think we need to test for window - if we want to be able to use yano in a pure node environment (where window maybe undefined).

Reply:
Moved window functions to the DOM variant. Assuming DOM functions aren't used in Node? Apologies for my ignorance, I've done very little pure node.